types/src/validator_verifier.rs

```diff
@@ -159,88 +185,112 @@ impl<PublicKey: VerifyingKey> ValidatorVerifier<PublicKey> {
         Ok(())
     }
 
-    /// Ensure there are at least quorum_size and not more than maximum expected signatures.
-    fn check_num_of_signatures<T>(
+    /// Ensure there is at least quorum_voting_power in the provided signatures and there
+    /// are only known authors. According to the threshold verification policy,
+    /// invalid public keys are not allowed.
+    pub fn check_voting_power_and_signatures<'a>(
         &self,
-        aggregated_signature: &HashMap<AccountAddress, T>,
-    ) -> std::result::Result<(), VerifyError>
-    where
-        T: Into<PublicKey::SignatureMaterial> + Clone,
-    {
-        let num_of_signatures = aggregated_signature.len();
-        if num_of_signatures < self.quorum_size {
-            return Err(VerifyError::TooFewSignatures {
-                num_of_signatures,
-                quorum_size: self.quorum_size,
-            });
+        authors: impl Iterator<Item = &'a AccountAddress>,
+        invalid_keys_allowed: bool,
```

Comment:
As discussed, maybe we want to avoid accepting invalid keys to prevent against adding arbitrary content.

Reply:
Yes, I'm adding this logic back.